### PR TITLE
Extend tracked backend functions with count metric

### PR DESCRIPTION
**Pull request by michalwski**
Before only function execution time was measured,
there was no information about number of given function calls
which is often as important as the execution time.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[6048.1](https://travis-ci.org/esl/MongooseIM/jobs/474398362) / Erlang 19.3 / small_tests / d446e09e2a4ab4abc590f0dc9288075183e3607c
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2177/6048/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2177/6048/small_tests.19.3/small/ct_run.mongooseim@localhost.2019-01-02_13.11.38/lib.mongooseim.logs/run.2019-01-02_13.11.39/suite.log.html)

File: src/backend_module.erl [hunk 1]
```diff
@@ -101,24 +101,28 @@ generate_fun(BaseModule, RealBackendModule, F, A, TrackedFuns) ->
 fun_header(F, Args) ->
     [atom_to_list(F), "(", Args, ")"].
 
--define(METRIC(Module, Op), [backends, Module, Op]).
+time_metric(Module, Op) -> [backends, Module, Op].
```
Comment (aleklisi) [inline src/backend_module.erl]: I would stay with macros as these function calls are replacements, but I don't insist ;)

File: src/backend_module.erl [hunk 2]
```diff
@@ -101,24 +101,28 @@ generate_fun(BaseModule, RealBackendModule, F, A, TrackedFuns) ->
...
-    %%     mongoose_metrics:update(global, ?METRIC(Backend, F), Time),
+    %%     mongoose_metrics:update(global, time_metric(Backend, F), Time),
     %%     Result.
-    ["    {Time, Result} = timer:tc(", RealBackendModule, ", ", FS, ", [", Args, "]), \n",
-     "    mongoose_metrics:update(global, ",
-     io_lib:format("~p", [?METRIC(BaseModule, F)]),
-     ", Time), \n",
+    CountMetric = io_lib:format("~p", [count_metric(BaseModule, F)]),
+    TimeMetric = io_lib:format("~p", [time_metric(BaseModule, F)]),
+    ["    mongoose_metrics:update(global, ", CountMetric, ", 1), \n",
```
Comment (aleklisi) [inline src/backend_module.erl]: What are whitespaces for?

Comment (aleklisi) [reply src/backend_module.erl]: Both a the beginning `["    mongoose_metrics:`... and at the end, before new line character   ...`1), \n",`

**Pull request merged**